api/v1beta1/agentserviceconfig_types.go

```diff
@@ -88,6 +88,15 @@ type AgentServiceConfigSpec struct {
 	// that are used if one the operators fails to be successfully deployed
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Must-Gather Images"
 	MustGatherImages []MustGatherImage `json:"mustGatherImages,omitempty"`
+
+	// IPXEHTTPRoute is controlling whether the operator is creating plain HTTP routes
+	// iPXE hosts may not work with router cyphers and may access artifacts via HTTP only
+	// This setting defaults to false, accepts "enabled,disabled". Empty value defaults to disabled
+	// The following endpoints would be exposed via http:
+	// * api/assisted-installer/v2/infra-envs/<id>/downloads/files?file_name=ipxe-script in assisted-service
+	// * boot-artifacts/ and images/<infra-enf id>/pxe-initrd in -image-service
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Expose IPXE HTTP route"
```

Comment:
I think we also want:
```
// +kubebuilder:validation:Enum=enabled;disabled
// +kubebuilder:validation:default:=disabled
```

https://sdk.operatorframework.io/docs/building-operators/golang/references/openapi-validation/